iwyu_output.cc

```diff
@@ -367,14 +368,33 @@ string PrintForwardDeclare(const NamedDecl* decl,
     if (const RecordDecl* rec = DynCastFrom(ctx)) {
       fwd_decl = std::string(rec->getName()) + "::" + fwd_decl;
     } else if (const NamespaceDecl* ns = DynCastFrom(ctx)) {
+      bool first = !seen_namespace;
       if (!seen_namespace) {
         seen_namespace = true;
         fwd_decl = tpl_params_and_kind + " " + fwd_decl;
       }
 
-      const std::string ns_name = ns->isAnonymousNamespace() ?
-          "" : (std::string(ns->getName()) + " ");
-      fwd_decl = "namespace " + ns_name + "{ " + fwd_decl + " }";
+      if (cxx17ns) {
+        const std::string ns_name = ns->isAnonymousNamespace() ?
+            "" : (std::string(ns->getName()) + "");
+        std::string prefix = ns_name;
+        std::string suffix;
+        if (first) {
+          first = false;
+          prefix = prefix + " { ";
+        }
+        if (ctx->getParent() && isa<NamedDecl>(ctx->getParent())) {
```

Comment:
Agreed, I don't think it can be anything else in practice, but NamedDecl is consistent with the above check in the for loop, so sounds like a good idea to keep it as-is.